Shared/src/com/pacoapp/paco/shared/util/SearchUtil.java

```diff
@@ -0,0 +1,237 @@
+        
+      }
+    }
+
+    if(node instanceof InExpression){
+      if ((((InExpression) node).getLeftExpression().getClass().getName().contains("Column"))&& node.toString().contains(columnName)){
+        String listWithParen = ((InExpression)node).getRightItemsList().toString().replace('(', ' ');
+        String listWithoutParen = listWithParen.replace(')', ' ' );
+        String[] arr = listWithoutParen.split(", ");
+        queriedValueList.addAll(Arrays.asList(arr));
+      }
+    }
+    if(node instanceof IsNullExpression){
+      if ((((IsNullExpression) node).getLeftExpression().getClass().getName().contains("Column"))&& node.toString().contains(columnName)){
+        queriedValueList.add("isnull");
+      }
+    }
+    
+  }    
+  
+  public static void getColumnNamesInWhere(Expression node, List<String> colNames){
+    getColumnNames(node, colNames);
+  }
+  
+  public static void getColumnNamesInHaving(Expression node, List<String> colNames){
+    getColumnNames(node, colNames);
+  }
+  
+  public static void getColumnNamesInSortOrder(String clause,  List<String> colNames){
+    getColumnNames(clause, colNames);
+  }
+  public static void getColumnNamesInGroupBy(String clause,  List<String> colNames){
+    getColumnNames(clause, colNames);
+  }
+  
+  private static net.sf.jsqlparser.statement.Statement getJsqlStatement(String selectSql){
+    net.sf.jsqlparser.statement.Statement statement = null;
+    try {
+      statement = CCJSqlParserUtil.parse(selectSql);
+    } catch (JSQLParserException e) {
+      // TODO Auto-generated catch block
+      e.printStackTrace();
+    }
+    List<String> colNames = Lists.newArrayList(); 
+    
+    Select selectStatement = (Select) statement;
+    return selectStatement;
+  }
+  
+  public static List<String> getAllColNamesInQuery(String selectSql){
+
+    List<String> colNames = Lists.newArrayList(); 
+    Select selectStatement = (Select)getJsqlStatement(selectSql);
+    PlainSelect pl = (PlainSelect)selectStatement.getSelectBody();
+    pl.getStringList(colNames);
+    for(SelectItem s: pl.getSelectItems()){
+      colNames.add(s.toString());
+    }
+    BinaryExpression where = (BinaryExpression) pl.getWhere();
+    getColumnNamesInWhere(where, colNames);
+    BinaryExpression having = (BinaryExpression) pl.getHaving();
+    getColumnNamesInHaving(having, colNames);
+    if(pl.getOrderByElements()!=null){
+      getColumnNamesInSortOrder(pl.getOrderByElements().toString(), colNames);
+    }
+    if(pl.getGroupByColumnReferences()!=null){
+      getColumnNamesInGroupBy(pl.getGroupByColumnReferences().toString(), colNames);
+    }
+    return colNames;
+    
+  }
+  
+  public static  String identifyTablesInvolved(List<String> colNamesInQuery){
+    if(colNamesInQuery.contains("text") || colNamesInQuery.contains("answer")){
+      return "eventsoutputs";
+    }else{
+      return "events";
+    }
+      
+  }
+  
+  public static  String getPlainSql(SQLQuery sqlQuery) {
```

Comment:
Not sure, but would it be better/possible to use jsqlparser's builders to construct a Select from the sqlQuery and then run toString on it?